kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -152,6 +165,59 @@ void prometheusMetricsWithCommonTags(KafkaCluster cluster) {
         }
     }
 
+    @Test
+    void testKroxyliciousInboundDownstreamMessages(KafkaCluster cluster, Topic topic) {
```

Comment:
nit: we are not terribly consistent on this but we are favouring omitting the word test.   We are also trying to make the test name describe the use-case.  

testKroxyliciousInboundDownstreamMessages -> canCountInboundDownstreamRecords.

Reply:
As @k-wall says, we try to use `intention revealing names` in general and particularly in tests where its very much use-cases or scenario descriptions. Its been organic growth so you will probably find lots of `testXXX` methods floating about. 

My junit5 Test Method template is 
```
@Test
void should${NAME}() {
    // Given

    // When

    // Then
}
```

So `shouldIncrementDownstreamMessagesOnProduceRequest` yes it gets long but when it fails its really clear to the person reading the failure what the problematic behaviour is.